api/auth.py

```diff
@@ -0,0 +1,41 @@
+"""JWT authenticator decorator for endpoints."""
+import json
+import jwt
+from user_actions import user_exists
+
+
+def jwt_auth(handler_function):
+    """Authenticates JWT against User DB"""
+
+    def _require_auth(self, *args, **kwargs):
+        token = self.request.headers.get_list('Authorization').pop()
+        if token:
+            parts = token.split()
+            if parts[0].lower() != 'bearer':
+                throw_authorization_error(self)
+            elif len(parts) == 1:
```

Comment:
This is an interesting pattern. You could have done `elif len(parts) >= 1` and gotten rid of the next `elif` but we get a slightly more fine-grained error if it is thrown on line 17 (we know there was exactly 1 as opposed to an unbounded number). Never really considered that before...